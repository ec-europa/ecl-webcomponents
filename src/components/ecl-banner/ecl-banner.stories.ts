const getArgs = () => ({
  variant: 'plain-background',
  bannerTitle: 'Headline sed elit lorem.',
  description: 'Pellentesque tempor tincidunt quam, finibus vulputate eros iaculis pharetra orci arcu, dictum maximus arcu pellentesque eget. Cras massa nunc.',
  ctaLabel: 'Subscribe',
  ctaLink: '/example.html',
  image: 'https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg',
  fullWidth: false,
  size: 'm',
  centered: true,
});

const getArgTypes = () => ({
  variant: {
    type: { name: 'select' },
    options: ['plain-background', 'text-box', 'image-overlay', 'text-highlight'],
    description: 'Banner variants',
  },
  size: {
    type: { name: 'select' },
    options: ['s', 'm', 'l'],
    description: 'Banner size', 
  },
  bannerTitle: {
    name: 'title',
    type: { name: 'string' },
    description: 'Title of the banner',
  },
  description: {
    type: { name: 'string' },
    description: 'Banner description',
  },
  image: {
    type: { name: 'string' },
    description: 'Image path or url',
  },
  ctaLabel: {
    name: 'label of the CTA button',
    type: { name: 'string' },
    description: 'CTA Link label',
  },
  ctaLink: {
    name: 'link of the CTA link',
    type: { name: 'string' },
    description: 'Link path or url',
  },
  fullWidth: {
    name: 'full width banner',
    type: { name: 'boolean' },
    description: 'Extends the banner to the viewport size',
  },
  centered: {
    type: { name: 'boolean' },
    description: 'Centered banner content',
  },
});

export default {
  title: 'Components/banner',
};

const Template = args =>
  `<ecl-banner
  variant="${args.variant}"
  theme="${args.theme}"
  cta-link="${args.ctaLink}"
  cta-label="${args.ctaLabel}"
  image="${args.image}"
  banner-title="${args.bannerTitle}"
  credit="Â© Copyright or credit"
  centered=${args.centered}
  size="${args.size}"
>
  ${args.description}
</ecl-banner>`;

export const Banner = Template.bind({});
Banner.storyName = 'default';
Banner.args = getArgs();
Banner.argTypes = getArgTypes();
