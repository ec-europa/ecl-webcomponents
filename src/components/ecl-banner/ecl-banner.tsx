import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-banner',
  styleUrls: {
    ec: './build/styles/ecl-banner-ec.css',
    eu: './build/styles/ecl-banner-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclBanner {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() external: boolean = false;
  @Prop() variant: string = 'display';
  @Prop() fullWidth: boolean = false;
  @Prop() bannerTitle: string;
  @Prop() image: string;
  @Prop() ctaLabel: string;
  @Prop() ctaLink: string;
  @Prop() centered: boolean = true;
  @Prop() credit: string;
  @Prop() size: string = 'm';
  @Prop() ctaClicked: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-banner`,
      `ecl-banner--${this.variant}`,
      `ecl-banner--${this.size}`,
      this.styleClass
    ];

    if (this.centered) {
      styleClasses.push('ecl-banner--centered');
    }
    if (this.fullWidth) {
      styleClasses.push('ecl-banner--full-width');
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    this.el.querySelector('.ecl-banner__cta').addEventListener('click', () => {
      this.ctaClicked = true;
    });
  }

  render() {
    const imageBanners = [
      'text-box',
      'image-overlay',
      'text-highlight',
    ];
    return (
      <section class={this.getClass()}>
      { imageBanners.includes(this.variant) && this.image ?
        <div
          class="ecl-banner__image"
          style={{ backgroundImage: "url(" + this.image + ")"}}
        >
        </div>
        : '' }
      { this.credit && this.variant != 'plain-background' ?
        <div class="ecl-banner__credit">{this.credit}</div> : ''
      }
        <div class="ecl-container">
          <div class="ecl-banner__container">
            <div class="ecl-banner__content">
            { this.bannerTitle ?
              <div class="ecl-banner__title">
                <span class="ecl-banner__title-text">{this.bannerTitle}</span>
              </div> : ''
            }
              <p class="ecl-banner__description">
                <span class="ecl-banner__description-text">
                  <slot></slot>
                </span>
              </p>
            { this.ctaLabel && this.ctaLink ?
              <div class="ecl-banner__cta">
                <ecl-link
                  variant="cta"
                  path={this.ctaLink}
                  style-class={`ecl-banner__link-cta sc-ecl-banner-${this.theme}`}
                >
                  {this.ctaLabel}
                  <ecl-icon
                    icon="corner-arrow"
                    slot="icon-after"
                    transform="rotate-90"
                  ></ecl-icon>
                </ecl-link>
              </div>
            : '' }
            </div>
          </div>
        </div>
      </section>
    );
  }
}
