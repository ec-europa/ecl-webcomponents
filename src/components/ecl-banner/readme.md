# ecl-banner



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type      | Default     |
| ------------- | -------------- | ----------- | --------- | ----------- |
| `bannerTitle` | `banner-title` |             | `string`  | `undefined` |
| `centered`    | `centered`     |             | `boolean` | `false`     |
| `credit`      | `credit`       |             | `string`  | `undefined` |
| `ctaClicked`  | `cta-clicked`  |             | `boolean` | `false`     |
| `ctaLabel`    | `cta-label`    |             | `string`  | `undefined` |
| `ctaLink`     | `cta-link`     |             | `string`  | `undefined` |
| `external`    | `external`     |             | `boolean` | `false`     |
| `fullWidth`   | `full-width`   |             | `boolean` | `false`     |
| `image`       | `image`        |             | `string`  | `undefined` |
| `imageAlt`    | `image-alt`    |             | `string`  | `undefined` |
| `size`        | `size`         |             | `string`  | `'m'`       |
| `styleClass`  | `style-class`  |             | `string`  | `undefined` |
| `theme`       | `theme`        |             | `string`  | `'ec'`      |
| `variant`     | `variant`      |             | `string`  | `'display'` |


## Dependencies

### Used by

 - [ecl-carousel-item](../ecl-carousel)

### Depends on

- [ecl-picture](../ecl-picture)
- [ecl-link](../ecl-link)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-banner --> ecl-picture
  ecl-banner --> ecl-link
  ecl-banner --> ecl-icon
  ecl-link --> ecl-icon
  ecl-carousel-item --> ecl-banner
  style ecl-banner fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
