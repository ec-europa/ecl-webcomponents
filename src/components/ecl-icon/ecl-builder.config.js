const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
