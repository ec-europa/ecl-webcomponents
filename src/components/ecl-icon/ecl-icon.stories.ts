import iconPathEc from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
import iconPathEu from '@ecl/resources-eu-icons/dist/sprites/icons.svg';
import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

export default {
  title: 'Components/icon',
};

const Template = (args) => `<ecl-icon 
                              icon="${args.icon}"
                              color="${args.color}"
                              size="${args.size}"
                              transform="${args.transform}"
                              path="${args.theme === 'ec' ? iconPathEc : iconPathEu}"
                            ></ecl-icon>`;

export const Icon = Template.bind({});
Icon.storyName = 'default';
Icon.args = {
  theme: 'ec',
  icon: 'close',
  transform: '',
  color: 'default',
  size: '2xl',
};
Icon.argTypes = {
  theme: {
    type: { name: 'select'},
    options: ['ec', 'eu'],
  },
  icon: {
    type: { name: 'select' },
    options: iconsAllEc,
    description: "Add an icon",
  },
  size: {
    type: { name: 'select' },
    options: ['2xs','xs', 's', 'm', 'l', 'xl', '2xl'],
    description: "icon size",
  },
  color: {
    type: { name: 'select' },
    options: ['default', 'primary', 'inverted'],
    description: "icon color",
  },
  transform: {
    type: { name: 'select' },
    description: 'Button icon transform',
    options: [
      'rotate-90',
      'rotate-180',
      'rotate-270',
      'flip-horizontal',
      'flip-vertical',
    ],
  },
};