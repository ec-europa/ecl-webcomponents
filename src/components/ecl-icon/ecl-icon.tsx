import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}