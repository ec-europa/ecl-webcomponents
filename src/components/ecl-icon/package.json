{
  "name": "@ecl/web-components-icon",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.12.0",
  "description": "ECL Icon",
  "main": "ecl-icon.ts",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:copy": "cross-env ecl-builder copy",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/resources-ec-icons": "3.13.0",
    "@ecl/resources-eu-icons": "3.13.0",
    "@ecl/specs-component-icon": "3.13.0",
    "@ecl/vanilla-component-icon": "3.13.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
