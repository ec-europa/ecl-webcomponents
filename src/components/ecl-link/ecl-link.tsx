import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
