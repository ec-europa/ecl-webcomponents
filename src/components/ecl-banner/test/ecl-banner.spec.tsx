import { newSpecPage } from '@stencil/core/testing';
import { EclBanner } from '../ecl-banner';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-banner', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclBanner, EclLink, EclIcon],
      html: `<ecl-banner
              variant="primary"
              theme="ec"
              cta-link="/example.html"
              cta-label="Subscribe"
              image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg"
              banner-title="EU Budget for the future"
              credit="Â© Copyright or credit"
              centered
            >
              Innovation, economy, environment and geopolitics
            </ecl-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
