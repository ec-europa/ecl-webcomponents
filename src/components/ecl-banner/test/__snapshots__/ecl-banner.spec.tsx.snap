// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-banner renders 1`] = `
<ecl-banner banner-title="EU Budget for the future" centered="" credit="© Copyright or credit" cta-label="Subscribe" cta-link="/example.html" image="https://inno-ecl.s3.amazonaws.com/media/examples/example.html-image.jpg" theme="ec" variant="primary">
  <!---->
  <section class="ecl-banner ecl-banner--centered ecl-banner--m ecl-banner--primary">
    <div class="ecl-banner__credit">
      © Copyright or credit
    </div>
    <div class="ecl-container">
      <div class="ecl-banner__container">
        <div class="ecl-banner__content">
          <div class="ecl-banner__title">
            <span class="ecl-banner__title-text">
              EU Budget for the future
            </span>
          </div>
          <p class="ecl-banner__description">
            <span class="ecl-banner__description-text">
              Innovation, economy, environment and geopolitics
            </span>
          </p>
          <div class="ecl-banner__cta">
            <ecl-link style-class="ecl-banner__link-cta sc-ecl-banner-ec">
              <!---->
              <a class="ecl-banner__link-cta ecl-link ecl-link--cta ecl-link--icon ecl-link--icon-after sc-ecl-banner-ec" href="/example.html">
                <ecl-icon slot="icon-after">
                  <svg class="ecl-icon ecl-icon--rotate-90 ecl-icon--xs ecl-link__icon ecl-link__icon--after">
                    <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
                  </svg>
                </ecl-icon>
                Subscribe
                <slot-fb hidden="" name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </div>
        </div>
      </div>
    </div>
  </section>
</ecl-banner>
`;
